File: src/RNetPi.Core/RNet/SetAllPowerPacket.cs
using System;

namespace RNetPi.Core.RNet;

/// <summary>
/// Packet for setting all zones power
/// </summary>
public class SetAllPowerPacket : EventPacket
{
    public SetAllPowerPacket(bool power)
    {
        TargetPath = new byte[] { 0x02, 0x00 };
        TargetControllerID = CONTROLLER_ALL;
        TargetZoneID = CONTROLLER_ALL;
        EventID = power ? (ushort)0xDD : (ushort)0xDC;
        EventTimestamp = 0;
        EventData = 0;
        EventPriority = 1;
    }

    public bool GetPower()
    {
        return EventID == 0xDD;
    }
}

File: src_js/rnet/packets/SetAllPowerPacket.js
const EventPacket = require("./EventPacket");

module.exports = class SetAllPowerPacket extends EventPacket {
    constructor(power) {
        super();

        this.targetPath = [0x02, 0x00];
        this.targetControllerID = 0x7E;
        this.eventID = 0xDD;
        this.eventData = 0x00;
        this.eventTimestamp = (power === true ? 1 : 0) << 8;
        this.eventPriority = 1;
    }
}
